packages/wallet/src/services/DelegationTracker/DelegationTracker.ts

```diff
@@ -71,19 +71,50 @@ export const certificateTransactionsWithEpochs = (
     )
   );
 
-export const createDelegationPortfolioTracker = (delegationTransactions: Observable<TxWithEpoch[]>) =>
-  delegationTransactions.pipe(
-    map((transactionsWithEpochs) => {
-      const txSorted = transactionsWithEpochs.sort((lhs, rhs) => lhs.epoch - rhs.epoch);
-      const latestDelegation = txSorted.pop();
-      if (!latestDelegation || !latestDelegation.tx.auxiliaryData || !latestDelegation.tx.auxiliaryData.blob)
-        return null;
+const hasDelegationCert = (certificates: Array<Cardano.Certificate> | undefined): boolean => {
+  if (!certificates || certificates.length === 0) return false;
 
-      const portfolio = latestDelegation.tx.auxiliaryData.blob.get(Cardano.DelegationMetadataLabel);
+  return certificates.some((cert) => {
+    let hasCert = false;
 
-      if (!portfolio) return null;
+    switch (cert.__typename) {
+      case Cardano.CertificateType.StakeDelegation:
+      case Cardano.CertificateType.StakeKeyRegistration:
+      case Cardano.CertificateType.StakeKeyDeregistration:
+        hasCert = true;
+        break;
+      default:
+        hasCert = false;
+    }
+
+    return hasCert;
+  });
+};
+
+export const createDelegationPortfolioTracker = (transactions: Observable<Cardano.HydratedTx[]>) =>
+  transactions.pipe(
+    map((hydratedTxs) => {
+      const sortedTransactions = [...hydratedTxs].sort((lhs, rhs) => rhs.blockHeader.slot - lhs.blockHeader.slot);
```

Comment:
It's already sorted, so here's a possible optimization. Or even just a `for` going in reverse
```suggestion
      const sortedTransactions = [...hydratedTxs].reverse();
```